File: playground/python/application/use_cases/create_order.py
"""Create order use case."""

from __future__ import annotations

from playground.python.domain.entities.order import Order
from playground.python.domain.value_objects.money import Money
from playground.python.domain.value_objects.order_id import OrderId


def create_order(order_id: OrderId, total: Money) -> Order:
    """Create a new order."""
    return Order(order_id=order_id, total=total)

File: playground/typescript/src/application/use_cases/CreateOrder.ts
import { Order } from "../../domain/entities/Order.js";
import type { Money } from "../../domain/value_objects/Money.js";
import type { OrderId } from "../../domain/value_objects/OrderId.js";

/** Create order use case. */
export function createOrder(orderId: OrderId, total: Money): Order {
  return new Order(orderId, total);
}
